main.py

```diff
@@ -0,0 +1,178 @@
+    parser.add_argument('device_name', type=str,
+                        help='name of the device as found in the first' +
+                        ' column in the /etc/fstab')
+    return parser
+
+
+def main():
+    """Main funtion"""
+    # init arguments
+    parser = init_arguments()
+    # process arguments
+    args = parser.parse_args()
+    if len(args.device_name) == 0:
+        print("Missing device name argument")
+        sys.exit(1)
+    with open("/etc/fstab", "r") as file:
+        entries = read_fstab(file.readlines())
+    found = False
+    for entry in entries:
+        if entry.device == args.device_name:
+            found = True
+            process_entry(entry)
+    if not found:
+        print('device '+args.device_name+' not found')
+
+
+def process_entry(entry):
+    """Process entry from fstab"""
+    if contains_tag(entry):
+        device_name = get_device_name(entry.device)
+        current_size_in_bytes = get_current_volume_size(device_name)
+        expected_size, expected_size_in_bytes = parse_tag(entry)
+        if current_size_in_bytes > expected_size_in_bytes:
+            if not is_block_device(entry.device):
+                raise BlockDeviceException('device ' + entry.device +
+                                           ' is not a block device')
+            if is_device_mounted(entry.device):
+                raise MountException('device '+entry.device+' is mounted')
+            shrink_volume(device_name, expected_size)
+        elif current_size_in_bytes < expected_size_in_bytes:
+            print('volume ' + entry.device + ' is already smaller than' +
+                  ' the expected size')
+
+
+def get_device_name(device):
+    """Determine device name. This can come handy when the device
+    name in the fstab is defined as a UUID instead of a path in /dev"""
+    if device.startswith('UUID='):
+        device_path = os.readlink("/dev/disk/by-uuid/"+device[len('UUID='):])
+        return "/dev"+os.path.realpath(device_path)
+    return device
+
+
+def get_tag_value(entry):
+    """Returns the value of the x-systemd.shrinkfs if defined in the options"""
+    for tag in entry.options.split(","):
+        if tag.startswith(SHRINK_TAG+'='):
+            return tag[len(SHRINK_TAG+'='):]
+    return ''
+
+
+def contains_tag(entry):
+    """Returns true if the x-systemd.shrinkfs option is defined for
+    the device entry in the fstab"""
+    return get_tag_value(entry) != ''
+
+
+def parse_tag(entry):
+    """Returns the value as captured in the tag and a representation of
+    the value in bytes
+    Eg: 1M -> (1M,1048576)"""
+    value = get_tag_value(entry)
+    # expect the unit to be the last char
+    return value, int(format_bytes(value[:len(value)-1], value[len(value)-1:]))
+
+
+def is_device_mounted(device_name):
+    """Returns true if the device is mounted."""
+    return subprocess.call(["/usr/bin/findmnt", "--source", device_name]) == 0
+
+
+def is_block_device(device_name):
```

Comment:
I think that `is_lvm` better describes what the function does

Reply:
Renamed